### PR TITLE
Flang extensions

### PR DESCRIPTION
**Pull request by grypp**
The patch generates wrapper device functions for Flang.

### CODE REVIEW

File: libomptarget/deviceRTLs/nvptx/src/libcall.cu [hunk 1]
```diff
@@ -460,3 +460,66 @@ EXTERN void xlf_debug_print_double(double *p) {
...
+}
+
+EXTERN int omp_get_cancellation_(void) { return omp_get_cancellation(); }
+
+EXTERN int omp_get_dynamic_(void) { return omp_get_dynamic(); }
+
+EXTERN void omp_lib_osnt4_(int num) { omp_set_num_threads(num); }
+
+EXTERN void omp_lib_osmal4_(int level) { omp_set_max_active_levels(level); }
+#endif
```
Comment (simoatze) [inline libomptarget/deviceRTLs/nvptx/src/libcall.cu]: Add new line here, if you build the runtime within LLVM tree I believe it complains with some warning.

**Review by simoatze — CHANGES_REQUESTED** [review]
Guray, this looks good, just a minor change and then I'll approve it.

File: libomptarget/deviceRTLs/nvptx/src/libcall.cu (hunk 1 above)
Comment (grypp) [reply libomptarget/deviceRTLs/nvptx/src/libcall.cu]: Thanks! I am going to add it.

Comment (sscalpone) [reply libomptarget/deviceRTLs/nvptx/src/libcall.cu]: Hi Güray, have you updated this yet?

Comment (grypp) [reply libomptarget/deviceRTLs/nvptx/src/libcall.cu]: Fixed

**Comment by grypp** [comment]
I changed the interface of Fortran wrapper library functions from pass-by-value to reference as Fortran always pass arguments by reference.

**Pull request merged**